packages/messaging/src/generic.ts

```diff
@@ -52,6 +52,12 @@ export interface GenericMessenger<
    * @param data The data to send with the message.
    * @param args Different messengers will have additional arguments to configure how a message gets sent.
    */
+  sendMessage<TType extends keyof TProtocolMap>(
+    type: TType,
+    ...args: GetDataType<TProtocolMap[TType]> extends undefined
+      ? [undefined?, ...TSendMessageArgs]
+      : never
+  ): Promise<GetReturnType<TProtocolMap[TType]>>;
```

Comment:
Interesting, I tried to do this before, but couldn't get it to work. Wonder what's different about your implementation... it's too long ago to remember what I did, but it probably has something to do with the `never` you're using.